src/libstd/ffi/c_str.rs

```diff
@@ -200,6 +200,28 @@ impl CString {
         CString { inner: v }
     }
 
+    /// Retakes ownership from a Rust string that has been returned to C.
+    pub unsafe fn from_raw_ptr(ptr: *const libc::c_char) -> CString {
+        let len = libc::strlen(ptr);
+        let len_with_nul = len as usize + 1;
+        let vec = Vec::from_raw_parts(ptr as *mut u8, len_with_nul, len_with_nul);
+        CString::from_vec_unchecked(vec)
+    }
```

Comment:
This function needsa  warning  itself that it only handles precisely the pointers returned from into_raw_ptr. Or otherwise explain the preconditions.